### ISSUE TITLE
Default typeRoots is prone to conflicts when combined with npm package hoisting

### ISSUE THREAD

**Issue by elliottsj** [microsoft/TypeScript#31373]
**TypeScript Version:**  3.4.5

**Search Terms:** typeroots npm hoisting

**Code**

```shell
$ npm install typescript@3.4.5 fork-ts-checker-webpack-plugin@1.3.1
$ echo "const x: number = setTimeout(() => {}, 1000);" > index.ts
$ ./node_modules/.bin/tsc --noEmit ./index.ts
index.ts:1:7 - error TS2322: Type 'Timeout' is not assignable to type 'number'.

1 const x: number = setTimeout(() => {}, 1000);
        ~


Found 1 error.
```

**Expected behavior:**

No error.

**Actual behavior:**

An error is raised because TypeScript finds and uses the alternative `setTimeout` definition from `node_modules/@types/node`, which is hoisted to the root `node_modules` folder due to the transitive dependency:

```
├─┬ fork-ts-checker-webpack-plugin@1.3.1
│ ├─┬ @types/webpack@4.4.31
│ │ ├── @types/node@12.0.0
```

[Version 1.3.2 of fork-ts-checker-webpack-plugin no longer includes this dependency](https://github.com/Realytics/fork-ts-checker-webpack-plugin/pull/287), so the error is gone:

```shell
$ npm install fork-ts-checker-webpack-plugin@1.3.2
$ ./node_modules/.bin/tsc --noEmit ./index.ts  # should exit 0
```

However, the problem with moving `@types` packages to devDependencies is that the users of a package like fork-ts-checker-webpack-plugin now need to depend on these `@types` packages in the consuming project, and keep them up-to-date, which is  counter to the recommendation here: https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#dependencies

>All dependencies are managed by npm. Make sure all the declaration packages you depend on are marked appropriately in the `"dependencies"` section in your `package.json`.

Users can work around this issue by setting a custom `"typeRoots"` value, but I don't think this is ideal because this typeRoots config needs to be kept up-to-date, and there is a higher burden on library maintainers to tell users to configure typeRoots.

**Comment by RyanCavanaugh** [comment]
Looks like a duplicate of #18588
